solidity/core/contracts/Common.sol

```diff
@@ -20,12 +20,12 @@ abstract contract Common is ICommon, OwnableUpgradeable {
 
     // ============ Public Variables ============
 
-    // Checkpoints of root => leaf index
-    // Checkpoints of index 0 have to be disallowed as the existence of such
-    // a checkpoint cannot be distinguished from their non-existence
-    mapping(bytes32 => uint256) public checkpoints;
-    // The latest checkpointed root
-    bytes32 public checkpointedRoot;
+    // Cached checkpoints, mapping root => leaf index.
+    // Cached checkpoints must have index > 0 as the presence of such
+    // a checkpoint cannot be distinguished from its absence.
+    mapping(bytes32 => uint256) public cachedCheckpoints;
+    // The latest cached root
+    bytes32 public latestCachedRoot;
```

Comment:
I assumed it was useful for off-chain agents. Since the plan is only to cache on the Outbox, and only when slashing, I figured it's not a big deal to make that moderately more expensive.